plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/sessionconfig/CodeScanSessionConfig.kt

```diff
@@ -175,39 +180,53 @@ class CodeScanSessionConfig(
 
     fun getProjectPayloadMetadata(): PayloadMetadata {
         val files = mutableSetOf<String>()
+        val traversedDirectories = mutableSetOf<VirtualFile>()
         val stack = Stack<VirtualFile>()
         var currentTotalFileSize = 0L
         var currentTotalLines = 0L
         val languageCounts = mutableMapOf<CodeWhispererProgrammingLanguage, Int>()
 
-        stack.push(projectRoot)
-        while (stack.isNotEmpty()) {
-            val current = stack.pop()
-
-            if (!current.isDirectory) {
-                if (runBlocking { !featureDevSessionContext.ignoreFile(current, this) }) {
-                    if (willExceedPayloadLimit(currentTotalFileSize, current.length)) {
-                        break
+        project.modules.forEach { module ->
```

Comment:
test?

Reply:
Wouldn't exisiting create project Payload tests cover this? or do i need add one to confirm modules in project?